server.go

```diff
@@ -728,6 +729,10 @@ func (srv *Server) readUDP(conn *net.UDPConn, timeout time.Duration) ([]byte, *S
 
 // WriteMsg implements the ResponseWriter.WriteMsg method.
 func (w *response) WriteMsg(m *Msg) (err error) {
+	if w.closed {
+		panic("dns: WriteMsg called after Close")
```

Comment:
not a superfan of all these panics

Reply:
better IHMO to just return Err{err: "text"} on these

Reply:
Ack.